mitosheet/src/components/taskpanes/UpdateImports/UpdateImportsTaskpane.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { useState } from "react";
+import MitoAPI from "../../../jupyter/api";
+import { AnalysisData, SheetData, UIState, UserProfile } from "../../../types"
+import TextButton from "../../elements/TextButton";
+import { DataframeImportParams } from "../../import/DataframeImportScreen";
+import DefaultTaskpane from "../DefaultTaskpane/DefaultTaskpane";
+import DefaultTaskpaneBody from "../DefaultTaskpane/DefaultTaskpaneBody";
+import DefaultTaskpaneFooter from "../DefaultTaskpane/DefaultTaskpaneFooter";
+import DefaultTaskpaneHeader from "../DefaultTaskpane/DefaultTaskpaneHeader";
+import { CSVImportParams } from "../../import/CSVImportScreen";
+import { ExcelImportParams } from "../../import/XLSXImportScreen";
+import ImportCard from "./UpdateImportCard";
+import { useStateFromAPIAsync } from "../../../hooks/useStateFromAPIAsync";
+import { getBaseOfPath, isCSVImportData, isCSVImportParams, isExcelImportData, isExcelImportParams } from "./UpdateImportsUtils";
+import { ImportScreen } from "../FileImport/FileImportTaskpane";
+import UpdateDataframeImportTaskpane from "./UpdateDataframeImportTaskpane";
+import UpdateXLSXImportsTaskpane from "./UpdateXLSXImportTaskpane";
+import UpdateCSVImportTaskpane from "./UpdateCSVImportTaskpane";
+import UpdateFileBrowserTaskpane from "./UpdateFileBrowserTaskpane";
+
+
+interface updateImportsTaskpaneProps {
+    mitoAPI: MitoAPI;
+    userProfile: UserProfile;
+    analysisData: AnalysisData;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+
+    currPathParts: string[];
+    setCurrPathParts: (newCurrPathParts: string[]) => void;
+
+    sheetDataArray: SheetData[];
+    selectedSheetIndex: number;
+}
+
+export interface CSVImportData {
+    step_type: 'simple_import',
+    params: CSVImportParams
+}
+export interface ExcelImportData {
+    step_type: 'excel_import',
+    params: ExcelImportParams
+}
+export interface DataframeImportData {
+    step_type: 'dataframe_import',
+    params: DataframeImportParams
+}
+
+export interface StepImportData {
+    step_id: string,
+    imports: (CSVImportData | ExcelImportData | DataframeImportData)[];
+}
+
+export interface DataframeCreationIndex {step_id: string, index: number}
+export interface ReplacingDataframeState {
+    dataframeCreationIndex: DataframeCreationIndex, 
+    screen: ImportScreen | 'dataframe_import',
```

Comment:
Since there is now a difference between the import screen and the update screens, maybe we should separate them here too.